pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -963,7 +973,106 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1.Mac
 		return &clustererror.RequeueAfterError{RequeueAfter: time.Second * 5}
 	}
 
-	// node is now available again
+	// node is now available again - restore annotations and labels
+	if _, annotationsBackupExists := machine.Annotations[nodeAnnotationsBackupAnnotation]; annotationsBackupExists {
+		return a.restoreAnnotationsAndLabels(node, machine, ctx)
+	}
+
+	//remediation is done
 	log.Printf("Node %s is available, remediation of Machine %s complete", node.Name, machine.Name)
 	return a.deleteRemediationAnnotations(ctx, machine)
 }
+
+// storeAnnotationsAndLabels copies node's annotations and labels and stores them on machine's annotations
+func (a *Actuator) storeAnnotationsAndLabels(node *corev1.Node, machine *machinev1.Machine, ctx context.Context) error {
+	marshaledAnnotations, err := marshal(node.Annotations)
+	if err != nil {
+		log.Printf("Failes to marshal node %s annotations assoicated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return err
+	}
+
+	marshaledLabels, err := marshal(node.Labels)
+	if err != nil {
+		log.Printf("Failes to marshal node %s labels assoicated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return err
+	}
+
+	machine.Annotations[nodeLabelsBackupAnnotation] = marshaledLabels
+	machine.Annotations[nodeAnnotationsBackupAnnotation] = marshaledAnnotations
+
+	err = a.client.Update(ctx, machine)
+	if err != nil {
+		log.Printf("failed to update machine with node's annotations and labels %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	return &clustererror.RequeueAfterError{}
+}
+
+// restoreAnnotationsAndLabels copies annotations and labels stored in machine's annotation to the node
+// and removes the annotations used to store that data
+func (a *Actuator) restoreAnnotationsAndLabels(node *corev1.Node, machine *machinev1.Machine, ctx context.Context) error {
+	if len(machine.Annotations) == 0 {
+		return &clustererror.RequeueAfterError{}
+	}
+
+	nodeAnn, err := unmarshal(machine.Annotations[nodeAnnotationsBackupAnnotation])
+	if err != nil {
+		log.Printf("failed to unmarshal node's annotations from %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	nodeLabels, err := unmarshal(machine.Annotations[nodeLabelsBackupAnnotation])
+	if err != nil {
+		log.Printf("failed to unmarshal node's labels from %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	node.Annotations = nodeAnn
+	node.Labels = nodeLabels
+
+	if err := a.client.Update(ctx, node); err != nil {
+		log.Printf("failed to update machine with node's annotations %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	delete(machine.Annotations, nodeAnnotationsBackupAnnotation)
+	delete(machine.Annotations, nodeLabelsBackupAnnotation)
+
+	err = a.client.Update(ctx, machine)
+	if err != nil {
+		log.Printf("failed to remove node %s annotations backup from machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return err
+	}
+
+	return &clustererror.RequeueAfterError{}
```

Comment:
Why force a re-queue here?

Reply:
I'm not sure if `reconcile()` get triggered for changes done by itself. 
If it's not - it makes sense to requeue, since we just restored annotations, and there are additional steps to perform, which are not triggered by any object change that will trigger `requeue()

Reply:
Any event from the resource will trigger reconciliation, regardless of who or what caused the event.

Reply:
Got it. Removed requeue.